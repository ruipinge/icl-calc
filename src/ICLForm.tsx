import React from 'react';
import { Formik, Form, Field, ErrorMessage } from 'formik';
import { differenceInYears } from 'date-fns';

import { FieldWithUnit } from './components/FieldWithUnit';
import { ICLSchema } from './ICLSchema';
import {
  calcIclSphere,
  calcIclCylindre,
  calcIclAxis,
  calcIclCylindreEquivalent
} from './formulas';

interface Patient {
  name: string;
  dateOfBirth: string;
  eye: 'left' | 'right';
}

interface Biometrics {
  ata: number;
  wtw: number;
  clr: number;
  acq: number;
  acan: number;
  acat: number;
  kf: number;
  cct: number;
}

interface SpectacleRefraction {
  sphere: number;
  cylindre: number;
  axis: number;
  vertex: number;
}

export interface ICLInputs {
  patient: Patient;
  biometrics: Biometrics;
  spectacleRefraction: SpectacleRefraction;
}

interface ContainerProps {
  initialValues: ICLInputs;
  setStatus: (arg1: boolean) => void;
}

export const ICLForm: React.FC<ContainerProps> = ({
  initialValues,
  setStatus
}) => {
  return (
    <Formik
      initialValues={initialValues}
      validationSchema={ICLSchema}
      onSubmit={(values, { setSubmitting }) => {
        setTimeout(() => {
          alert(JSON.stringify(values, null, 2));
          setSubmitting(false);
        }, 400);
      }}
    >
      {({ errors, touched, values }) => (
        <Form>
          <h2>Patient</h2>
          <div className="form-row">
            <div className="form-group col-md-4">
              <label htmlFor="fieldName">Name</label>
              <Field
                name="patient.name"
                className="form-control"
                id="fieldName"
                placeholder="enter patient name"
                autoComplete="off"
              />
            </div>
            <div className="form-group col-md-2 offset-md-1">
              <label htmlFor="fieldDateOfBirth">Date of Birth</label>
              <Field
                name="patient.dateOfBirth"
                className={
                  (errors.patient?.dateOfBirth && touched.patient?.dateOfBirth
                    ? 'is-invalid'
                    : '') + ' form-control'
                }
                id="fieldDateOfBirth"
                placeholder="yyyy-mm-dd"
                autoComplete="off"
                maxLength={10}
              />
              <ErrorMessage
                name="patient.dateOfBirth"
                component="div"
                className="invalid-feedback"
              />
            </div>
            <div className="form-group col-md-1 offset-md-1">
              <label htmlFor="fieldAge">Age</label>
              <div className="input-group">
                <input
                  name="patient.age"
                  className="form-control text-right"
                  id="fieldAge"
                  disabled={true}
                  value={
                    values.patient.dateOfBirth && !errors.patient?.dateOfBirth
                      ? differenceInYears(
                          new Date(),
                          new Date(values.patient.dateOfBirth)
                        ) || ''
                      : ''
                  }
                />
                <div className="input-group-append">
                  <span className="input-group-text" title="years">
                    yr.
                  </span>
                </div>
              </div>
            </div>
            <div className="form-group col-md-2 offset-md-1">
              <label htmlFor="fieldEye">Eye</label>
              <Field
                as="select"
                name="eye"
                className="form-control"
                id="fieldEye"
                autoComplete="off"
              >
                <option value="">Select...</option>
                <option value="left">Left</option>
                <option value="right">Right</option>
              </Field>
            </div>
          </div>
          <hr />
          <div className="form-row">
            <div className="col-md-4">
              <h2>Biometrics</h2>
              <FieldWithUnit
                label="Angle to Angle (AtA)"
                name="biometrics.ata"
                unit="mm"
                error={errors.biometrics?.ata}
              />
              <FieldWithUnit
                label="White to White (WtW)"
                name="biometrics.wtw"
                unit="mm"
                unitTitle="millimetres"
                error={errors.biometrics?.wtw}
              />
              <FieldWithUnit
                label="Crystaline Lens Rise (CLR)"
                name="biometrics.clr"
                unit="nm"
                unitTitle="nanometres"
                error={errors.biometrics?.clr}
              />
              <FieldWithUnit
                label="Internal Anterior Chamber Depth (ACQ)"
                name="biometrics.acq"
                unit="mm"
                unitTitle="millimetres"
                error={errors.biometrics?.acq}
              />
              <FieldWithUnit
                label="Anterior Chamber Angle nasal (ACAn)"
                name="biometrics.acan"
                unit="º"
                unitTitle="degrees"
                error={errors.biometrics?.acan}
              />
              <FieldWithUnit
                label="Anterior Chamber Angle temporal (ACAt)"
                name="biometrics.acat"
                unit="º"
                unitTitle="degrees"
                error={errors.biometrics?.acat}
              />
              <FieldWithUnit
                label="Keratometry - Flat Meridian (Kf)"
                name="biometrics.kf"
                unit="dpt"
                unitTitle="dioptres"
                error={errors.biometrics?.kf}
              />
              <FieldWithUnit
                label="Central Corneal Thickness (CCT)"
                name="biometrics.cct"
                unit="μm"
                unitTitle="micrometres"
                error={errors.biometrics?.cct}
              />
            </div>
            <div className="col-md-3 offset-md-1">
              <h2>Spectacle Refraction</h2>
              <FieldWithUnit
                label="Sphere"
                name="spectacleRefraction.sphere"
                unit="dpt"
                unitTitle="dioptres"
                error={errors.spectacleRefraction?.sphere}
              />
              <FieldWithUnit
                label="Cylindre"
                name="spectacleRefraction.cylindre"
                unit="dpt"
                unitTitle="dioptres"
                error={errors.spectacleRefraction?.cylindre}
              />
              <FieldWithUnit
                label="Axis"
                name="spectacleRefraction.axis"
                unit="º"
                unitTitle="degrees"
                error={errors.spectacleRefraction?.axis}
              />
              <FieldWithUnit
                label="Vertex"
                name="spectacleRefraction.vertex"
                unit="mm"
                unitTitle="millimetres"
                error={errors.spectacleRefraction?.vertex}
              />
            </div>
            <div className="col-md-3 offset-md-1">
              <h2>ICL Power</h2>
              <FieldWithUnit
                label="Sphere"
                name="iclSphere"
                value={calcIclSphere({
                  sphere: values.spectacleRefraction.sphere,
                  cylindre: values.spectacleRefraction.cylindre
                })}
                unit="dpt"
                unitTitle="dioptres"
                disabled={true}
              />
              <FieldWithUnit
                label="Cylindre"
                name="iclCylindre"
                value={calcIclCylindre({
                  sphere: values.spectacleRefraction.sphere,
                  cylindre: values.spectacleRefraction.cylindre
                })}
                unit="dpt"
                unitTitle="dioptres"
                disabled={true}
              />
              <FieldWithUnit
                label="Axis"
                name="iclAxis"
                value={calcIclAxis(values.spectacleRefraction.axis)}
                unit="º"
                unitTitle="degrees"
                disabled={true}
              />
              <FieldWithUnit
                label="Spherical Equivalent"
                value={calcIclCylindreEquivalent({
                  sphere: values.spectacleRefraction.sphere,
                  cylindre: values.spectacleRefraction.cylindre
                })}
                name="iclSphericalEquivalent"
                unit="º"
                unitTitle="degrees"
                disabled={true}
              />
            </div>
          </div>
        </Form>
      )}
    </Formik>
  );
};
