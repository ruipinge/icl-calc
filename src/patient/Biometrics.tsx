import { FieldWithUnit } from './FieldWithUnit';
import { FormikState } from 'formik';
import { ICLInputs } from '../types';

export const Biometrics = ({
  errors,
  touched
}: FormikState<Pick<ICLInputs, 'biometrics'>>) => (
  <>
    <h4>Biometrics</h4>
    <FieldWithUnit
      label="Angle to Angle (AtA)"
      name="biometrics.ata"
      unit="mm"
      error={errors.biometrics?.ata}
      touched={touched.biometrics?.ata}
    />
    <FieldWithUnit
      label="White to White (WtW)"
      name="biometrics.wtw"
      unit="mm"
      error={errors.biometrics?.wtw}
      touched={touched.biometrics?.wtw}
    />
    <FieldWithUnit
      label="Crystaline Lens Rise (CLR)"
      name="biometrics.clr"
      unit="μm"
      error={errors.biometrics?.clr}
      touched={touched.biometrics?.clr}
    />
    <FieldWithUnit
      label="Internal Anterior Chamber Depth (ACD)"
      name="biometrics.acd"
      unit="mm"
      error={errors.biometrics?.acd}
      touched={touched.biometrics?.acd}
    />
    <FieldWithUnit
      label="Anterior Chamber Angle nasal (ACAn)"
      name="biometrics.acan"
      unit="º"
      error={errors.biometrics?.acan}
      touched={touched.biometrics?.acan}
    />
    <FieldWithUnit
      label="Anterior Chamber Angle temporal (ACAt)"
      name="biometrics.acat"
      unit="º"
      error={errors.biometrics?.acat}
      touched={touched.biometrics?.acat}
    />
    <FieldWithUnit
      label="Keratometry - Flat Meridian (Kf)"
      name="biometrics.kf"
      unit="dpt"
      error={errors.biometrics?.kf}
      touched={touched.biometrics?.kf}
    />
    <FieldWithUnit
      label="Central Corneal Thickness (CCT)"
      name="biometrics.cct"
      unit="μm"
      error={errors.biometrics?.cct}
      touched={touched.biometrics?.cct}
    />
  </>
);
