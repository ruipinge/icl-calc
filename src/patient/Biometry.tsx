import { FieldWithUnit } from './FieldWithUnit';
import { FormikState } from 'formik';
import { ICLInputs } from '../types';

export const Biometry = ({
  errors,
  touched
}: FormikState<Pick<ICLInputs, 'biometry'>>) => (
  <>
    <h4>Biometry</h4>
    <FieldWithUnit
      label="Angle to Angle (AtA)"
      name="biometry.ata"
      unit="mm"
      error={errors.biometry?.ata}
      touched={touched.biometry?.ata}
    />
    <FieldWithUnit
      label="White to White (WtW)"
      name="biometry.wtw"
      unit="mm"
      error={errors.biometry?.wtw}
      touched={touched.biometry?.wtw}
    />
    <FieldWithUnit
      label="Crystaline Lens Rise (CLR)"
      name="biometry.clr"
      unit="μm"
      error={errors.biometry?.clr}
      touched={touched.biometry?.clr}
    />
    <FieldWithUnit
      label="Internal Anterior Chamber Depth (ACD)"
      name="biometry.acd"
      unit="mm"
      error={errors.biometry?.acd}
      touched={touched.biometry?.acd}
    />
    <FieldWithUnit
      label="Anterior Chamber Angle nasal (ACAn)"
      name="biometry.acan"
      unit="º"
      error={errors.biometry?.acan}
      touched={touched.biometry?.acan}
    />
    <FieldWithUnit
      label="Anterior Chamber Angle temporal (ACAt)"
      name="biometry.acat"
      unit="º"
      error={errors.biometry?.acat}
      touched={touched.biometry?.acat}
    />
    <FieldWithUnit
      label="Keratometry - Flat Meridian (Kf)"
      name="biometry.kf"
      unit="dpt"
      error={errors.biometry?.kf}
      touched={touched.biometry?.kf}
    />
    <FieldWithUnit
      label="Central Corneal Thickness (CCT)"
      name="biometry.cct"
      unit="μm"
      error={errors.biometry?.cct}
      touched={touched.biometry?.cct}
    />
  </>
);
