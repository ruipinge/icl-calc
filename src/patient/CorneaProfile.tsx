import { Field, FormikState } from 'formik';
import { ICLInputs, PreviousSurgery } from '../types';

import { FieldWithUnit } from './FieldWithUnit';

export const CorneaProfile = ({
  errors,
  touched
}: FormikState<Pick<ICLInputs, 'corneaProfile'>>) => (
  <div className="section-form">
    <h4>Cornea Profile</h4>
    <FieldWithUnit
      label="Anterior Keratometry Flat (KAntFlt)"
      name="corneaProfile.kaf"
      unit="dpt"
      error={errors.corneaProfile?.kaf}
      touched={touched.corneaProfile?.kaf}
    />
    <FieldWithUnit
      label="@"
      name="corneaProfile.axisaf"
      unit="º"
      error={errors.corneaProfile?.axisaf}
      touched={touched.corneaProfile?.axisaf}
    />
    <FieldWithUnit
      label="Anterior Keratometry Steep (KAntStp)"
      name="corneaProfile.kas"
      unit="dpt"
      error={errors.corneaProfile?.kas}
      touched={touched.corneaProfile?.kas}
    />
    <FieldWithUnit
      label="@"
      name="corneaProfile.axisas"
      unit="º"
      error={errors.corneaProfile?.axisas}
      touched={touched.corneaProfile?.axisas}
    />
    <FieldWithUnit
      label="Posterior Keratometry Flat (KPostFlt)"
      name="corneaProfile.kpf"
      unit="dpt"
      error={errors.corneaProfile?.kpf}
      touched={touched.corneaProfile?.kpf}
    />
    <FieldWithUnit
      label="@"
      name="corneaProfile.axispf"
      unit="º"
      error={errors.corneaProfile?.axispf}
      touched={touched.corneaProfile?.axispf}
    />
    <FieldWithUnit
      label="Posterior Keratometry Steep (KPostStp)"
      name="corneaProfile.kps"
      unit="dpt"
      error={errors.corneaProfile?.kps}
      touched={touched.corneaProfile?.kps}
    />
    <FieldWithUnit
      label="@"
      name="corneaProfile.axisps"
      unit="º"
      error={errors.corneaProfile?.axisps}
      touched={touched.corneaProfile?.axisps}
    />
    <FieldWithUnit
      label="Central Corneal Thickness (CCT)"
      name="corneaProfile.cct"
      unit="μm"
      error={errors.corneaProfile?.cct}
      touched={touched.corneaProfile?.cct}
    />
    <div className={'form-group row mb-4'}>
      <label
        htmlFor={'corneaProfile.previousSurgery'}
        className="col-sm-6 col-form-label"
      >
        Previous Corneal Refractive Surgery
      </label>
      <div className="col-sm-6">
        <div className="input-group">
          <Field
            as="select"
            name="corneaProfile.previousSurgery"
            className="form-control"
          >
            {Object.values(PreviousSurgery).map((val) => (
              <option value={val} key={val}>
                {val}
              </option>
            ))}
          </Field>
        </div>
      </div>
    </div>
  </div>
);
