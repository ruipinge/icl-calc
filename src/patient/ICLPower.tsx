import {
  calcIclAxis,
  calcIclCylindre,
  calcIclCylindreEquivalent,
  calcIclSphere
} from '../formulas';

import { FieldWithUnit } from './FieldWithUnit';
import { FormikState } from 'formik';
import { ICLInputs } from '../types';

export const ICLPower = ({
  errors,
  touched,
  values
}: FormikState<Pick<ICLInputs, 'spectacleRefraction'>>) => (
  <>
    <h4>ICL Power</h4>
    <FieldWithUnit
      label="Sphere"
      name="iclSphere"
      value={calcIclSphere({
        sphere: values.spectacleRefraction.sphere,
        cylindre: values.spectacleRefraction.cylindre
      })}
      unit="dpt"
      disabled={true}
    />
    <FieldWithUnit
      label="Cylindre"
      name="iclCylindre"
      value={calcIclCylindre({
        sphere: values.spectacleRefraction.sphere,
        cylindre: values.spectacleRefraction.cylindre
      })}
      unit="dpt"
      disabled={true}
    />
    <FieldWithUnit
      label="Axis"
      name="iclAxis"
      value={calcIclAxis(values.spectacleRefraction.axis)}
      unit="ยบ"
      disabled={true}
    />
    <FieldWithUnit
      label="Spherical Equivalent"
      value={calcIclCylindreEquivalent({
        sphere: values.spectacleRefraction.sphere,
        cylindre: values.spectacleRefraction.cylindre
      })}
      name="iclSphericalEquivalent"
      unit="ยบ"
      disabled={true}
    />
  </>
);
