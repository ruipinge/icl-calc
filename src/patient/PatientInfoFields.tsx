import { Field, ErrorMessage, FormikState } from 'formik';
import { differenceInYears } from 'date-fns';

import { getClassName } from '../util';

export interface PatientInfo {
  name: string;
  dateOfBirth: string;
  eye: 'left' | 'right';
}

export const calcAge = ({
  dateOfBirth,
  error
}: {
  dateOfBirth?: string;
  error?: string;
}): number =>
  (dateOfBirth &&
    !error &&
    differenceInYears(new Date(), new Date(dateOfBirth))) ||
  0;

export const PatientInfoFields = ({
  errors,
  values,
  touched
}: FormikState<PatientInfo | undefined>) => (
  <>
    <h2>Patient</h2>
    <div className="form-row">
      <div className="form-group col-md-4">
        <label htmlFor="fieldName">Name</label>
        <Field
          name="patient.name"
          className="form-control"
          id="fieldName"
          placeholder="enter patient name"
          autoComplete="off"
        />
      </div>
      <div className="form-group col-md-2 offset-md-1">
        <label htmlFor="fieldDateOfBirth">Date of Birth</label>
        <Field
          name="patient.dateOfBirth"
          className={getClassName({
            error: errors?.dateOfBirth,
            touched: touched?.dateOfBirth
          })}
          id="fieldDateOfBirth"
          placeholder="yyyy-mm-dd"
          autoComplete="off"
          maxLength={10}
        />
        <ErrorMessage
          name="patient.dateOfBirth"
          component="div"
          className="invalid-feedback"
        />
      </div>
      <div className="form-group col-md-2">
        <label htmlFor="fieldAge">Age</label>
        <div className="input-group">
          <input
            name="age"
            className="form-control text-right"
            id="fieldAge"
            disabled={true}
            value={calcAge({
              dateOfBirth: values?.dateOfBirth,
              error: errors?.dateOfBirth
            })}
          />
          <div className="input-group-append">
            <span className="input-group-text" title="years">
              yr.
            </span>
          </div>
        </div>
      </div>
      <div className="form-group col-md-2 offset-md-1">
        <label htmlFor="fieldEye">Eye</label>
        <Field
          as="select"
          name="patient.eye"
          className="form-control"
          id="fieldEye"
        >
          <option value="">Select...</option>
          <option value="left">Left</option>
          <option value="right">Right</option>
        </Field>
      </div>
    </div>
  </>
);
