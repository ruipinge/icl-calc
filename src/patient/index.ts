export interface PatientInfo {
  name: string;
  dateOfBirth: string;
  eye: 'left' | 'right';
}

export interface Biometrics {
  ata: number;
  wtw: number;
  clr: number;
  acq: number;
  acan: number;
  acat: number;
  kf: number;
  cct: number;
}

export interface SpectacleRefraction {
  sphere: number;
  cylindre: number;
  axis: number;
  vertex: number;
}

export interface ICLInputs {
  patient: PatientInfo;
  biometrics: Biometrics;
  spectacleRefraction: SpectacleRefraction;
}

export const INITIAL_VALUES: ICLInputs = {
  patient: {
    name: '',
    dateOfBirth: '',
    eye: 'left'
  },
  biometrics: {
    ata: 0,
    wtw: 0,
    clr: 0,
    acq: 0,
    acan: 0,
    acat: 0,
    kf: 0,
    cct: 0
  },
  spectacleRefraction: {
    sphere: 0,
    cylindre: 0,
    axis: 0,
    vertex: 0
  }
};
