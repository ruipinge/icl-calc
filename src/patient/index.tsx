import { Biometry } from './Biometry';
import { CorneaProfile } from './CorneaProfile';
import { FormikState } from 'formik';
import { ICLInputs } from '../types';
import { ICLPower } from './ICLPower';
import { Info } from './Info';
import { SpectacleRefraction } from './SpectacleRefraction';

export const Patient = ({
  errors,
  touched,
  values,
  ...otherProps
}: FormikState<ICLInputs>) => (
  <>
    <Info values={values} errors={errors} touched={touched} {...otherProps} />
    <hr />
    <div className="form-row">
      <div className="col-md-3">
        <Biometry
          values={values}
          errors={errors}
          touched={touched}
          {...otherProps}
        />
      </div>
      <div className="col-md-4 offset-md-1">
        <CorneaProfile
          values={values}
          errors={errors}
          touched={touched}
          {...otherProps}
        />
      </div>
      <div className="col-md-3 offset-md-1">
        <SpectacleRefraction
          values={values}
          errors={errors}
          touched={touched}
          {...otherProps}
        />
        <ICLPower
          values={values}
          errors={errors}
          touched={touched}
          {...otherProps}
        />
      </div>
    </div>
  </>
);
