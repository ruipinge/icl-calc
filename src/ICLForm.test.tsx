import TestRenderer from 'react-test-renderer';

import { INITIAL_VALUES } from './patient';
import { ICLForm } from './ICLForm';

it('renders without crashing', () => {
  const tree = TestRenderer.create(
    <ICLForm initialValues={INITIAL_VALUES} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
