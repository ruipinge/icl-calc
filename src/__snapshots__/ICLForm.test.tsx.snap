// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <h2>
    Patient
  </h2>
  <div
    className="form-row"
  >
    <div
      className="form-group col-md-4"
    >
      <label
        htmlFor="fieldName"
      >
        Name
      </label>
      <input
        autoComplete="off"
        className="form-control"
        id="fieldName"
        name="patient.name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="enter patient name"
        value=""
      />
    </div>
    <div
      className="form-group col-md-2 offset-md-1"
    >
      <label
        htmlFor="fieldDateOfBirth"
      >
        Date of Birth
      </label>
      <input
        autoComplete="off"
        className="form-control"
        id="fieldDateOfBirth"
        maxLength={10}
        name="patient.dateOfBirth"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="yyyy-mm-dd"
        value=""
      />
    </div>
    <div
      className="form-group col-md-2"
    >
      <label
        htmlFor="fieldAge"
      >
        Age
      </label>
      <div
        className="input-group"
      >
        <input
          className="form-control text-right"
          disabled={true}
          id="fieldAge"
          name="age"
          value={0}
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
            title="years"
          >
            yr.
          </span>
        </div>
      </div>
    </div>
    <div
      className="form-group col-md-2 offset-md-1"
    >
      <label
        htmlFor="fieldEye"
      >
        Eye
      </label>
      <select
        className="form-control"
        id="fieldEye"
        name="patient.eye"
        onBlur={[Function]}
        onChange={[Function]}
        value="left"
      >
        <option
          value=""
        >
          Select...
        </option>
        <option
          value="left"
        >
          Left
        </option>
        <option
          value="right"
        >
          Right
        </option>
      </select>
    </div>
  </div>
  <hr />
  <div
    className="form-row"
  >
    <div
      className="col-md-4"
    >
      <h2>
        Biometrics
      </h2>
      <div
        className="form-group row"
      >
        <label
          className="col-sm-6 col-form-label"
          htmlFor="biometrics.atafield"
        >
          Angle to Angle (AtA)
        </label>
        <div
          className="col-sm-6"
        >
          <div
            className="input-group"
          >
            <input
              autoComplete="off"
              className="form-control text-right"
              name="biometrics.ata"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value={0}
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                title="millimetres"
              >
                mm
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-group row"
      >
        <label
          className="col-sm-6 col-form-label"
          htmlFor="biometrics.wtwfield"
        >
          White to White (WtW)
        </label>
        <div
          className="col-sm-6"
        >
          <div
            className="input-group"
          >
            <input
              autoComplete="off"
              className="form-control text-right"
              name="biometrics.wtw"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value={0}
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                title="millimetres"
              >
                mm
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-group row"
      >
        <label
          className="col-sm-6 col-form-label"
          htmlFor="biometrics.clrfield"
        >
          Crystaline Lens Rise (CLR)
        </label>
        <div
          className="col-sm-6"
        >
          <div
            className="input-group"
          >
            <input
              autoComplete="off"
              className="form-control text-right"
              name="biometrics.clr"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value={0}
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                title="nanometres"
              >
                nm
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-group row"
      >
        <label
          className="col-sm-6 col-form-label"
          htmlFor="biometrics.acqfield"
        >
          Internal Anterior Chamber Depth (ACQ)
        </label>
        <div
          className="col-sm-6"
        >
          <div
            className="input-group"
          >
            <input
              autoComplete="off"
              className="form-control text-right"
              name="biometrics.acq"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value={0}
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                title="millimetres"
              >
                mm
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-group row"
      >
        <label
          className="col-sm-6 col-form-label"
          htmlFor="biometrics.acanfield"
        >
          Anterior Chamber Angle nasal (ACAn)
        </label>
        <div
          className="col-sm-6"
        >
          <div
            className="input-group"
          >
            <input
              autoComplete="off"
              className="form-control text-right"
              name="biometrics.acan"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value={0}
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                title="degrees"
              >
                º
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-group row"
      >
        <label
          className="col-sm-6 col-form-label"
          htmlFor="biometrics.acatfield"
        >
          Anterior Chamber Angle temporal (ACAt)
        </label>
        <div
          className="col-sm-6"
        >
          <div
            className="input-group"
          >
            <input
              autoComplete="off"
              className="form-control text-right"
              name="biometrics.acat"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value={0}
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                title="degrees"
              >
                º
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-group row"
      >
        <label
          className="col-sm-6 col-form-label"
          htmlFor="biometrics.kffield"
        >
          Keratometry - Flat Meridian (Kf)
        </label>
        <div
          className="col-sm-6"
        >
          <div
            className="input-group"
          >
            <input
              autoComplete="off"
              className="form-control text-right"
              name="biometrics.kf"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value={0}
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                title="dioptres"
              >
                dpt
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-group row"
      >
        <label
          className="col-sm-6 col-form-label"
          htmlFor="biometrics.cctfield"
        >
          Central Corneal Thickness (CCT)
        </label>
        <div
          className="col-sm-6"
        >
          <div
            className="input-group"
          >
            <input
              autoComplete="off"
              className="form-control text-right"
              name="biometrics.cct"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value={0}
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                title="micrometres"
              >
                μm
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-md-3 offset-md-1"
    >
      <h2>
        Spectacle Refraction
      </h2>
      <div
        className="form-group row"
      >
        <label
          className="col-sm-6 col-form-label"
          htmlFor="spectacleRefraction.spherefield"
        >
          Sphere
        </label>
        <div
          className="col-sm-6"
        >
          <div
            className="input-group"
          >
            <input
              autoComplete="off"
              className="form-control text-right"
              name="spectacleRefraction.sphere"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value={0}
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                title="dioptres"
              >
                dpt
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-group row"
      >
        <label
          className="col-sm-6 col-form-label"
          htmlFor="spectacleRefraction.cylindrefield"
        >
          Cylindre
        </label>
        <div
          className="col-sm-6"
        >
          <div
            className="input-group"
          >
            <input
              autoComplete="off"
              className="form-control text-right"
              name="spectacleRefraction.cylindre"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value={0}
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                title="dioptres"
              >
                dpt
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-group row"
      >
        <label
          className="col-sm-6 col-form-label"
          htmlFor="spectacleRefraction.axisfield"
        >
          Axis
        </label>
        <div
          className="col-sm-6"
        >
          <div
            className="input-group"
          >
            <input
              autoComplete="off"
              className="form-control text-right"
              name="spectacleRefraction.axis"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value={0}
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                title="degrees"
              >
                º
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-group row"
      >
        <label
          className="col-sm-6 col-form-label"
          htmlFor="spectacleRefraction.vertexfield"
        >
          Vertex
        </label>
        <div
          className="col-sm-6"
        >
          <div
            className="input-group"
          >
            <input
              autoComplete="off"
              className="form-control text-right"
              name="spectacleRefraction.vertex"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value={0}
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                title="millimetres"
              >
                mm
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-md-3 offset-md-1"
    >
      <h2>
        ICL Power
      </h2>
      <div
        className="form-group row"
      >
        <label
          className="col-sm-6 col-form-label"
          htmlFor="iclSpherefield"
        >
          Sphere
        </label>
        <div
          className="col-sm-6"
        >
          <div
            className="input-group"
          >
            <input
              className="form-control text-right"
              disabled={true}
              name="iclSphere"
              value={-1.34}
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                title="dioptres"
              >
                dpt
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-group row"
      >
        <label
          className="col-sm-6 col-form-label"
          htmlFor="iclCylindrefield"
        >
          Cylindre
        </label>
        <div
          className="col-sm-6"
        >
          <div
            className="input-group"
          >
            <input
              className="form-control text-right"
              disabled={true}
              name="iclCylindre"
              value={0}
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                title="dioptres"
              >
                dpt
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-group row"
      >
        <label
          className="col-sm-6 col-form-label"
          htmlFor="iclAxisfield"
        >
          Axis
        </label>
        <div
          className="col-sm-6"
        >
          <div
            className="input-group"
          >
            <input
              className="form-control text-right"
              disabled={true}
              name="iclAxis"
              value={90}
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                title="degrees"
              >
                º
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-group row"
      >
        <label
          className="col-sm-6 col-form-label"
          htmlFor="iclSphericalEquivalentfield"
        >
          Spherical Equivalent
        </label>
        <div
          className="col-sm-6"
        >
          <div
            className="input-group"
          >
            <input
              className="form-control text-right"
              disabled={true}
              name="iclSphericalEquivalent"
              value={-1.34}
            />
            <div
              className="input-group-append"
            >
              <span
                className="input-group-text"
                title="degrees"
              >
                º
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;
